InvenTree/templates/js/translated/part.js

```diff
@@ -324,20 +324,39 @@ function editCategory(pk) {
  * Delete a PartCategory via the API
  */
 function deletePartCategory(pk, options={}) {
-    var url = `/api/part/category/${pk}/`;
+    var url = `/api/part/category/${pk}/?child-categories-action=${pk}&child-parts-action=${pk}`;
 
     var html = `
     <div class='alert alert-block alert-danger'>
     {% trans "Are you sure you want to delete this part category?" %}
-    <ul>
-        <li>{% trans "Any child categories will be moved to the parent of this category" %}</li>
-        <li>{% trans "Any parts in this category will be moved to the parent of this category" %}</li>
-    </ul>
     </div>`;
-
+    var subChoices = [
+        {
+            value: 'move-to-parent',
+            display_name: '{% trans "Move under the parent category" %}',
+        },
+        {
+            value: 'delete',
+            display_name: '{% trans "Delete them" %}',
+        }
+    ];
     constructForm(url, {
         title: '{% trans "Delete Part Category" %}',
         method: 'DELETE',
+        fields: {
+            'child-parts-action': {
```

Comment:
Please do not use `-` characters here, as it makes field lookup difficult. `-` is generally not accepted in a variable name. Use `_` instead e.g. `child_parts_action`

Reply:
Yeah, I faced with it :)